### PR TITLE
Type Ratings | New Feature

### PR DESCRIPTION
**Pull request by FatihKoz**
Adds "Type Rating" definition and assignment possibility to v7 core.

- Adds a new setting
- Ability to define "Type Ratings"
- Ability to attach/detach type ratings to subfleets
- Ability to attach/detach type ratings to users

- User Service `getAllowableSubfleets()` now considers both Rank and Type Rating restrictions

- Adds a new vendor via composer for deep relationships

Also fixes a bug in SimBrief aircraft selection dropdown.

### CODE REVIEW

File: app/Database/migrations/2021_11_23_184532_add_type_rating_tables.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+use App\Contracts\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class AddTypeRatingTables extends Migration
+{
+    public function up()
+    {
+        if (!Schema::hasTable('typeratings')) {
```
Comment (nabeelio) [inline app/Database/migrations/2021_11_23_184532_add_type_rating_tables.php:11]: The table name and prefixes should be `type_ratings` and `type_rating_user`, `type_rating_subfleet`, etc. Just keeping with convention.

File: app/Database/seeds/settings.yml [hunk 2]
```diff
@@ -283,7 +283,14 @@
...
   type: boolean
-  description: 'Aircraft that can be flown are restricted to a user''s rank'
+  description: 'Aircraft that can be flown are restricted to a user''s rank' 
+- key: pireps.restrict_aircraft_to_typerating
+  name: 'Restrict Aircraft by Type Ratings'
+  group: pireps
+  value: false
+  options: ''
+  type: boolean
+  description: 'Aircraft that can be flown are restricted to Type Ratings user holds'
```
Comment (nabeelio) [inline app/Database/seeds/settings.yml:293]: "Aircraft restricted to user type ratings"

File: app/Database/seeds/settings.yml [hunk 3]
```diff
@@ -283,7 +283,14 @@
   value: true
   options: ''
   type: boolean
-  description: 'Aircraft that can be flown are restricted to a user''s rank'
+  description: 'Aircraft that can be flown are restricted to a user''s rank'
```
Comment (nabeelio) [inline app/Database/seeds/settings.yml:286]: "Aircraft restricted to user's rank"

File: app/Database/migrations/2021_11_23_184532_add_type_rating_tables.php (hunk 1 above)
Comment (FatihKoz) [reply app/Database/migrations/2021_11_23_184532_add_type_rating_tables.php:11]: Model name is `Typerating` , this is why the table name uses `typeratings` and intermediates followed that same name.

When I first build this, model name was `TypeRating` and the table names was like you suggested, it caused some problem with relationships (even though I used proper foreign keys). To overcome those problems I removed the camel case naming.

But left the repository name as it was.

**Pull request merged**